section/algorithms.html

```diff
@@ -1,539 +1,686 @@
-      This request is non-interactive: implementations MUST NOT include an {{GrantRequest.interact}} and
-      the authorization server will return a {{Grant}} directly if successful.</p>
-    <div class="algorithm">
-      When asked to <dfn>send an incoming payment grant request</dfn>, given |websiteWalletAddressDetails:WalletAddressDetails|
-      perform the following steps.
-    <ol>
-        <li>
-          Let |userWalletDetails:WalletAddressDetails?| be the result of [=get user wallet=].
-        </li>
-        <li>If |userWalletDetails| is null, then return failure.</li>
-      <li>
-        Let |authServer:URL| be the result of running [=URL parser=] with |websiteWalletAddressDetails|'s {{WalletAddressDetails/authServer}} property.
-      </li>
-      <li>
-        If |authServer| [=url/scheme=] is not `https`, return failure.
-      </li>
-      <li>
-        <p>Construct a {{ GrantRequest }} |grantRequest: GrantRequest| as follows:</p>
+    <h3>Resource server operations (RS)</h3>
+    <section>
+      <h4>Create incoming payment</h4>
+      <p>The create incoming payment request refers
+        to the <a data-cite="oprs/operations/create-incoming-payment#">Open Payments create incoming payment</a>.</p>
+      <div class="algorithm">
+        When asked to <dfn>send a create incoming payment request</dfn>,
+        given |websiteWalletAddressDetails:WalletAddressDetails|, |amountValue:DOMString|,
+        |incomingPaymentGrant:Grant|, perform the following steps.
         <ol>
           <li>
-            Set |grantRequest|'s {{GrantRequest/client}} to |userWalletDetails|.{{WalletAddressDetails/id}}.
+            Let |resourceServer:URL| be the result of running [=URL parser=] with |websiteWalletAddressDetails|'s {{WalletAddressDetails/resourceServer}} property.
+          </li>
+          <li>
+            If |resourceServer| [=url/scheme=] is not `https`, return failure.
           </li>
           <li>
-            Let |incomingPaymentAccess| be a new {{ Access }} object with:
+            Let |incomingPaymentsUrl:URL| be the result of running [=URL parser=] with the string formed by concatenating |resourceServer| and "/incoming-payments".
+          </li>
+          <li>
+            <p>Let |incomingAmount:Amount| be a new {{Amount}} with:</p>
             <ul>
-              <li>{{Access/type}} set to `"incoming-payment"`</li>
-              <li>{{Access/actions}} set to `["create"]`</li>
-              <li>{{Access/identifier}} set to |websiteWalletAddressDetails|'s {{WalletAddressDetails/id}} property</li>
+              <li>{{Amount/value}} set to |amountValue|</li>
+              <li>{{Amount/assetCode}} set to |websiteWalletAddressDetails|'s {{WalletAddressDetails/assetCode}}</li>
+              <li>{{Amount/assetScale}} set to |websiteWalletAddressDetails|'s {{WalletAddressDetails/assetScale}}</li>
             </ul>
           </li>
```

Comment:
We don't use amount with incoming payments for Web Monetization, as the amount is not fixed per session. We continue using same incoming payment until it expires and as long as user is on the page sending payments.

Reply:
ack